<?xml version="1.0" encoding="utf-8"?>
<launch>
    <arg name="connected" default="false"/>
    <arg name="sim" default="true"/>
    <arg name="speed" default="1.0"/>
    <arg name="plot" default="false"/>
    <arg name="ctx" default="false"/>
    <param name="connection" value="$(arg connected)"/>
    <param name="print_ctx" value="$(arg ctx)"/>

    <include file="$(find kuka_lab_setup)/launch/load_kukalabsetup.launch"/>

    <rosparam file="$(find kuka_rsi_hw_interface)/config/controller_joint_names.yaml" command="load"/>
    <rosparam file="$(find kuka_rsi_hw_interface)/test/test_params_sim.yaml" command="load" if="$(arg sim)"/>
    <rosparam file="$(find kuka_rsi_hw_interface)/test/test_params.yaml" command="load" unless="$(arg sim)"/>
    <node name="kuka_hardware_interface" pkg="kuka_rsi_hw_interface" type="kuka_hardware_interface_node" respawn="false" output="screen" required="true"/>

    <rosparam file="$(find etasl_ros_control_cmd)/config/smach_peg.yaml" command="load" subst_value="true"/>
    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" args="joint_state_controller"/>
    <node name="controller_spawner_etasl" pkg="controller_manager" type="spawner" respawn="false" output="screen" args="--stopped etasl_controller_cmd"/>

    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>

    <node name="kuka_rsi_simulator" pkg="kuka_rsi_simulator" type="kuka_rsi_simulator" args="127.0.0.1 49152" if="$(arg sim)"/>

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find kuka_lab_setup)/config/zero_state_robotiq.rviz" required="true"/>

    <node name="gripper_controller" pkg="robotiq_2f_gripper_control" type="Robotiq2FGripperTcpNode.py" args="192.168.250.11" required="true" if="$(arg connected)"/>
    <node name="netft_publisher" pkg="netft_control" type="netft_node" args="--rate 250 --wrench --address 192.168.1.13 --topic_name etasl_controller_cmd/" required="true"  if="$(arg connected)"/>
    <node name="wrench_publisher" pkg="etasl_ros_control_cmd" type="force_publisher.py" required="true" unless="$(arg connected)"/>

    <node name="jointstate_grapher" pkg="etasl_ros_control_cmd" type="jointstate_grapher.py" required="true" if="$(arg plot)"/>
    <node name="lissajous_publisher" pkg="etasl_ros_control_cmd" type="lissajous_publisher.py" required="true"/>
    <node name="zivid_publisher" pkg="etasl_ros_control_cmd" type="zivid_publisher.py" required="true"/>
    <node name="etasl_pose_publisher" pkg="etasl_ros_control_cmd" type="etasl_pose_publisher.py" required="true"/>
    <node name="speed_pub" pkg="etasl_ros_control_cmd" type="speed_pub.py" required="true"> 
        <param name="gain_multiplier" value="$(arg speed)"/>
    </node>
    
    <node name="smach_viz" pkg="smach_viewer" type="smach_viewer.py" required="true"/>
    <node name="smach_peg" pkg="etasl_ros_control_cmd" type="smach_task.py" required="true"/>

</launch>